### DISCUSSION TITLE
Neotree positions

### DISCUSSION THREAD

**Discussion by cseickel** [folke/edgy.nvim#9] [General]
I noticed that your README config for neo-tree is using "position=right" and "position=top" even though they are all on the left, I assume that's to work around the default behavior of one window/source per position. Have you tried position=current instead?

Honestly, it feels like I have more issues around managing the sidebar aspect than anything else, and I don't even use it as a sidebar anymore! If I can simply strip out that logic and tell people to use edgy instead, that would really make my day. Is there anything I can do in neo-tree to make that a reality? I would be willing to add in a new position type called "edgy" and alter its behavior to fit better with what edgy needs.

**Comment by folke** [comment]
Hi!

That's indeed to work-around the one window stuff.

`position=current` doesn't work, since that would move the current window in the sidebar. Edgy will then detect that there is no main window and will create a new one. That's fine, but in case your main area has splits, one of the splits would be gone.

Ideally, an option like `new=true` would be great. Users could then open multiple neo-tree windows in a certain position. For edgy, the position doesn't matter, since I move the windows anyway.

In edgy's case a `new=true` would be perfect for this, since that `open()` function will only ever be called if there's no window yet that the `open()` function creates.

**Comment by cseickel** [comment]
Does edgy also take care of focus stealing prevention? As in if something like Telescope opens a file in neotree's window, will edgy stop that or move that window somewhere else and restore neotree in the desired location?

**Reply by folke** [reply]
Edgy will expunge the telescope window and if neo-tree re-opens (automatically or manually) it will be re-added to the layout
